From 016591382e9310e5d515734f88bc77b340aa2f08 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:53:00 -0400
Subject: [fs] ext4: Remove wait for unwritten extents in ext4_ind_direct_IO()

Message-id: <1414058125-4183-27-git-send-email-lczerner@redhat.com>
Patchwork-id: 98122
O-Subject: [RHEL 7.1 PATCH v2 026/171] ext4: Remove wait for unwritten extents in ext4_ind_direct_IO()
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Jan Kara <jack@suse.cz>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 38b8ff7db4d99ebfabaee81b8747877424d1cf28

 We don't have to wait for unwritten extent conversion in
 ext4_ind_direct_IO() as all writes that happened before DIO are
 flushed by the generic code and extent conversion has happened before
 we cleared PageWriteback bit.

 Reviewed-by: Zheng Liu <wenqing.lz@taobao.com>
 Signed-off-by: Jan Kara <jack@suse.cz>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/indirect.c b/fs/ext4/indirect.c
index c4c0b13..5c836c7 100644
--- a/fs/ext4/indirect.c
+++ b/fs/ext4/indirect.c
@@ -675,11 +675,6 @@ ssize_t ext4_ind_direct_IO(int rw, struct kiocb *iocb,
 
 retry:
  if (rw == READ && ext4_should_dioread_nolock(inode)) {
-  if (unlikely(atomic_read(&EXT4_I(inode)->i_unwritten))) {
-   mutex_lock(&inode->i_mutex);
-   ext4_flush_unwritten_io(inode);
-   mutex_unlock(&inode->i_mutex);
-  }
   /*
    * Nolock dioread optimization may be dynamically disabled
    * via ext4_inode_block_unlocked_dio(). Check inode's state
-- 
1.7.1